### PR TITLE
ISSUE-224 Return respective error messages for different subject and version not found cases.

### PR DESCRIPTION
**Pull request by satishd**
- Return respective error messages for different subject and version not found cases. 
- Added respective tests.

### CODE REVIEW

**Comment by satishd** [comment]
Closes #224

**Comment by michaelandrepearce** [comment]
@satishd I will look tomorrow when in work. I know I didn't handle / map all the errors as the confluent serdes makes no distinction really, but it is a good thing to sort

File: schema-registry/rest-service/src/main/java/com/hortonworks/registries/schemaregistry/webservice/ConfluentSchemaRegistryCompatibleResource.java [hunk 1]
```diff
@@ -201,18 +201,42 @@ public Response getSchemaVersion(@ApiParam(value = "subject", required = true)
                                              String versionId) {
         Response response;
         try {
+            SchemaVersionInfo schemaVersionInfo = null;
 
-            SchemaVersionInfo schemaVersionInfo = "latest".equals(versionId)
-                    ? schemaRegistry.getLatestSchemaVersionInfo(subject)
-                    : schemaRegistry.getSchemaVersionInfo(Long.valueOf(versionId));
+            if ("latest".equals(versionId)) {
```
Comment (michaelandrepearce) [inline schema-registry/rest-service/src/main/java/com/hortonworks/registries/schemaregistry/webservice/ConfluentSchemaRegistryCompatibleResource.java]: we should probably add some versionId validation and throw invalidVersionException() if not valid input range.
see here for logic of the id validation https://github.com/confluentinc/schema-registry/blob/8eb664dbc84b1c2db3666fa0771eeb0e0909f892/core/src/main/java/io/confluent/kafka/schemaregistry/rest/VersionId.java

Comment (michaelandrepearce) [inline schema-registry/rest-service/src/main/java/com/hortonworks/registries/schemaregistry/webservice/ConfluentSchemaRegistryCompatibleResource.java]: this should be on all paths that take versionId

File: schema-registry/rest-service/src/main/java/com/hortonworks/registries/schemaregistry/webservice/ConfluentSchemaRegistryCompatibleResource.java [hunk 2]
```diff
@@ -201,18 +201,42 @@ public Response getSchemaVersion(@ApiParam(value = "subject", required = true)
...
+                response = WSUtils.respondEntity(schemaVersionEntry, Response.Status.OK);
+            }
         } catch (SchemaNotFoundException ex) {
-            LOG.error("No schema version found with id [{}]", versionId, ex);
-            response = versionNotFoundError();
+            LOG.error("No schema found with subject [{}]", subject, ex);
+            response = subjectNotFoundError();
         } catch (Exception ex) {
             LOG.error("Encountered error while retrieving all subjects", ex);
             response = serverError();
```
Comment (michaelandrepearce) [inline schema-registry/rest-service/src/main/java/com/hortonworks/registries/schemaregistry/webservice/ConfluentSchemaRegistryCompatibleResource.java]: should probably return RestSchemaRegistryException for these, wdyt?

**Comment by michaelandrepearce** [comment]
I don't see any issues with what is here, e.g. nothing incorrect. 

There is probably a few more cases we should add if we want exception handling to match also identical to confluent. I have added some comments where should be thought about. This though can be in another PR don't see why all has to be done in one. So I'm a +1 for this to merge

File: schema-registry/rest-service/src/main/java/com/hortonworks/registries/schemaregistry/webservice/ConfluentSchemaRegistryCompatibleResource.java (hunk 2 above)
Comment (satishd) [inline schema-registry/rest-service/src/main/java/com/hortonworks/registries/schemaregistry/webservice/ConfluentSchemaRegistryCompatibleResource.java]: No plans to address these as they are specific errors form storage etc. Users would anyway see the error message on the reason of failure of the invocation in their error message. Idea of this PR is to handle respective errors for no schema/version found.

**Comment by satishd** [comment]
Thanks @michaelandrepearce for reviewing, addressed/replied to your comments.

**Pull request merged**